xapian-core/api/queryinternal.cc

```diff
@@ -1295,6 +1297,70 @@ QueryValueLE::get_description() const
     return desc;
 }
 
+PostList*
+QueryValueLT::postlist(QueryOptimiser* qopt, double factor) const
+{
+    LOGCALL(QUERY, PostList*, "QueryValueLT::postlist", qopt | factor);
+    if (factor != 0.0)
+	qopt->inc_total_subqs();
+    const Xapian::Database::Internal& db = qopt->db;
+    const string& lb = db.get_value_lower_bound(slot);
+    if (lb.empty()) {
+	// This should only happen if there are no values in this slot (which
+	// could be because the backend just doesn't support values at all).
+	// If there were values in the slot, the backend should have a
+	// non-empty lower bound, even if it isn't a tight one.
+	AssertEq(db.get_value_freq(slot), 0);
+	RETURN(NULL);
+    }
+    if (limit < lb) {
+	RETURN(NULL);
+    }
+    if (limit > db.get_value_upper_bound(slot)) {
+	// The range check isn't needed, but we do still need to consider
+	// which documents have a value set in this slot.  If this value is
+	// set for all documents, we can replace it with the MatchAll
+	// postlist, which is especially efficient if there are no gaps in
+	// the docids.
+	if (db.get_value_freq(slot) == db.get_doccount()) {
+	    RETURN(db.open_post_list(string()));
+	}
+    }
+    RETURN(new ValueLtPostList(&db, slot, limit));
+}
+// This is the same as QueryValueLE::serialise,
+// not sure if it should be the same
```

Comment:
No, it needs to be different - otherwise if you serialise OP_VALUE_LT then unserialise the result youl get OP_VALUE_LE out:

```
olly@gemse:~/git/xapian/xapian-bindings/python$ ./run-python-test 
Python 2.7.17 (default, Jan 19 2020, 19:54:54) 
[GCC 9.2.1 20200110] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import xapian
>>> q = xapian.Query(xapian.Query.OP_VALUE_LT, 0, "abc")
>>> str(q)
'Query(VALUE_LT 0 abc)'
>>> str(xapian.Query.unserialise(q.serialise()))
'Query(VALUE_LE 0 abc)'
>>> 
```

Reply:
You marked the conversation as resolved, which is really something you should leave to the reviewer.

But while you have updated the serialisation code you haven't removed the comment with your question from the patch, and that's not something useful to include - it's just confusing now.

Reply:
@BodaSadalla98 as mentioned, this comment shouldn't be here.